crates/router/src/core/mandate.rs

```diff
@@ -259,6 +260,25 @@ where
     Ok(resp)
 }
 
+#[instrument(skip(state))]
+pub async fn retrieve_mandates_list(
+    state: &AppState,
+    merchant_account: storage::MerchantAccount,
+    constraints: api_models::mandates::MandateListConstraints,
+) -> RouterResponse<Vec<api_models::mandates::MandateResponse>> {
+    let mandates = state
+        .store
+        .find_mandates_by_merchant_id(&merchant_account.merchant_id, constraints)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Unable to retrieve mandates")?;
+    let mandates_list = future::try_join_all(mandates.iter().map(|mandate| {
+        mandates::MandateResponse::from_db_mandate(state, mandate.clone(), &merchant_account)
+    }))
+    .await?;
```

Comment:
You should be able to use `into_iter()` instead of `iter()` and can avoid the clone.